### DISCUSSION TITLE
Keyboard accessibility - datagrid

### DISCUSSION THREAD

**Discussion by siddharth1903** [vmware/clarity#6258] [Q&A] [answered]
Hey Guys,
I was glancing over the vpat report on the website and am interested in knowing the status of the data grid component for keyboard accessibility using arrow keys. Is this something that is planned for in the near future?

Thanks in advance,
Sid

**Comment by mathisscott** [comment, marked as answer]
This is something we are currently working on in the `@cds/core` datagrid. It is fully navigable with arrow keys. We expect to deliver the Core datagrid in 2-3 months time.

Following on that, we will direct people who want improved datagrid accessibility to use the Clarity Core datagrid. We will not be updating the Clarity Angular (`@clr/angular`) datagrid.

**Reply by siddharth1903** [reply]
@mathisscott having used clarity angular for more than 2 years now, this is very disappointing to know. Teams will take time to migrate completely and it happens over a period of time with testing, regression etc. To cut off NFR support of clr angular is an extreme step and gives no window for consumers to adopt. Does this not fall under the 6 month support period? I was hoping only new features were going to be cut off!

**Reply by Jinnie** [reply]
@siddharth1903 what you are requesting is a new feature for the angular datagrid. We don't intend to add new features to Clarity Angular. Many accessibility related problems are on the architectural or design levels, so their reworking in Clarity Angular is too costly and brings high risks of regressions.
Clarity Core brings you the best solution to these types of problems. To be able to use Clarity Core components, you will not need to fully rework your product. Components can be imported and replaced one by one, based on what you need and where you need it. Clarity Angular and Clarity Core can coexist in a project quite well. We are also working on adoption documentation and tools to make this process as seamless as possible.

**Reply by siddharth1903** [reply]
@Jinnie understood and thanks! There are different types of applications we run inside our organization. Certain adaptations that are already in production using Clr/angular , it will be difficult to convince clients that accessibility is a new feature. They are mostly considered as non functional's and usually are tied to wcag standards. These are huge/large scale projects that need a sizeable amount of investment to look at migrating into clr core entirely or partly, since data grid is our primary component and is used extensively in every screen. 
With the increasing number of lawsuits for ADA compliance, in the last years(around 10k in the U.S alone), it would be really helpful if the NFR promise is fulfilled on clr angular. I do understand the framework perspective though.

**Reply by siddharth1903** [reply]
@Jinnie a bit more on 
"To be able to use Clarity Core components, you will not need to fully rework your product. Components can be imported and replaced one by one, based on what you need and where you need it. Clarity Angular and Clarity Core can coexist in a project quite well."

Does the branding(css variables) automatically align between these two worlds if i have both clr angular and core components? Is there some documentation on this approach if we go down this path?
